sql/core/src/main/scala/org/apache/spark/sql/catalyst/util/V2ExpressionBuilder.scala

```diff
@@ -187,8 +187,9 @@ class V2ExpressionBuilder(e: Expression, isPredicate: Boolean = false) {
         assert(v.isInstanceOf[V2Predicate])
         new V2Not(v.asInstanceOf[V2Predicate])
       }
-    case UnaryMinus(child, true) => generateExpressionWithName("-", Seq(child))
-    case BitwiseNot(child) => generateExpressionWithName("~", Seq(child))
+    case unaryMinus @ UnaryMinus(_, true) =>
+      generateExpressionWithName("-", unaryMinus, isPredicate)
+    case _: BitwiseNot => generateExpressionWithName("~", expr, isPredicate)
     case CaseWhen(branches, elseValue) =>
```

Comment:
`CaseWhen` always seems to return `V2Predicate`, is this correct?

Reply:
I don't think so, but this probably doesn't hurt as `V2Predicate` extends `GeneralScalarExpression`. We should still fix it to make code clearer.

Reply:
Thanks, I made fix.